src/net.c

```diff
@@ -753,6 +759,198 @@ static int hyper_set_interface_mtu(struct rtnl_handle *rth,
 				IFLA_MTU);
 }
 
+static char* hyper_ifreq_to_mac_addr(struct ifreq ifr)
+{
+	char *mac_addr;
+
+	if (asprintf(&mac_addr,
+		     "%02x:%02x:%02x:%02x:%02x:%02x",
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[0],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[1],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[2],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[3],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[4],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[5]) == -1) {
+		fprintf(stderr, "failed to allocate mac_addr string\n");
+		return NULL;
+	}
+
+	return mac_addr;
+} 
+
+/*!
+ * Check hardware address related to the provided network interface name
+ * matches the expected hardware address. It expects EUI-48 MAC addresses.
+ *
+ * \param device network interface name.
+ * \param mac_addr hardware address (EUI-48) to match with \p device one.
+ *
+ * \note In case the function succeeds, it returns 0. In case the function
+ * fails, it returns -1.
+ */
+static int hyper_check_device_match_mac_addr(const char *device,
+					     const char *mac_addr)
+{
+	struct ifreq ifr;
+	int sock = -1;
+	char *tmp_mac_addr = NULL;
+	int ret = -1;
+
+	if (!device && !strcmp(device, "")) {
```

Comment:
This test and the `mac_addr` one below are incorrect and will likely cause a crash if `device == NULL`.

I'd tweak them to something like the following which saves the library call:

```
if (!device || !*device) {
```